src/bin/ion/commands/beta/generate/context.rs

```diff
@@ -75,14 +79,30 @@ pub enum AbstractDataType {
     Structure(bool),
 }
 
+impl AbstractDataType {
+    pub fn element_type(&self) -> Option<&String> {
```

Comment:
(🗺️ PR tour) Adds `element_type` and `sequence_type` APIs to get `Sequence` variant information from `AbstractDataType`.  These methods are used in `verify_and_update_abstract_data_type` which verifies that the abstract data type to be generated doesn't result into different type due to given ISL constraints.